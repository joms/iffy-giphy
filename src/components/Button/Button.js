import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ onClick, children, disabled, ...rest }) => {
    const click = e => {
        if (!disabled) {
            onClick(e);
        }
    };

    return (
        <button onClick={click} disabled={disabled} {...rest}>
            {children}
        </button>
    );
};

Button.defaultProps = {
    onClick: () => null,
    disabled: false,
};

Button.propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func,
    rest: PropTypes.object,
    disabled: PropTypes.bool,
};

export default Button;
