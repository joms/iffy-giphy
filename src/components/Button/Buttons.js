import React from 'react';
import PropTypes from 'prop-types';
import Button from '.';

const Buttons = ({ children }) => <div className="buttons">{children}</div>;

Buttons.propTypes = {
    children: PropTypes.node.isRequired,
};

export default Buttons;
