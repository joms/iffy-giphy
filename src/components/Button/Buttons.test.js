import React from 'react';
import { shallow } from 'enzyme';
import Button, { Buttons } from '.';

describe('<Buttons>', () => {
    const wrapper = shallow(
        <Buttons>
            <Button>Yolo</Button>
            <Button>Swag</Button>
        </Buttons>
    );
    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
