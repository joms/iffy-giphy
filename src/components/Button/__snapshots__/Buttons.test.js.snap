// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttons> matches snapshot 1`] = `
<div
  className="buttons"
>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    Yolo
  </Button>
  <Button
    disabled={false}
    onClick={[Function]}
  >
    Swag
  </Button>
</div>
`;
